libs/vkd3d/device.c

```diff
@@ -379,6 +379,184 @@ static bool vkd3d_disable_nvx_extensions(struct d3d12_device *device, const char
     return disabled;
 }
 
+static BOOL vkd3d_wait_vr_key(HKEY vr_key)
+{
+    DWORD type, value, wait_status, size;
+    LSTATUS status;
+    HANDLE event;
+    DWORD max_retry = 10; /* Each value query and each timeout counts as a try. */
+
+    size = sizeof(value);
+    if ((status = RegQueryValueExA(vr_key, "state", NULL, &type, (BYTE *)&value, &size)))
+    {
+        ERR("OpenVR: could not query value, status %d", status);
+        return false;
+    }
+    if (type != REG_DWORD)
+    {
+        ERR("OpenVR: unexpected value type %d", type);
+        return false;
+    }
+
+    if (value)
+        return value == 1;
+
+    event = CreateEventA(NULL, FALSE, FALSE, NULL);
+    while (max_retry)
+    {
+        if (RegNotifyChangeKeyValue(vr_key, FALSE, REG_NOTIFY_CHANGE_LAST_SET, event, TRUE))
+        {
+            ERR("Error registering registry change notification");
+            break;
+        }
+        size = sizeof(value);
+        if ((status = RegQueryValueExA(vr_key, "state", NULL, &type, (BYTE *)&value, &size)))
+        {
+            ERR("OpenVR: could not query value, status %d", status);
+            break;
+        }
+        if (value)
+            break;
+        max_retry--;
+
+        while ((wait_status = WaitForSingleObject(event, 1000)) == WAIT_TIMEOUT && max_retry)
```

Comment:
Is there a risk that we wait for a full second if steam.exe sets the registry entry after RegQuery but before RegNotifyChangeKeyValue?